scan.go

```diff
@@ -221,12 +242,29 @@ func (s *ScanService) Do() (*ScanCursor, error) {
 		path += "/" + strings.Join(typesPart, ",")
 	}
 
+	// Define the body of the request
+	var body map[string]interface{}
+
+	if s.source != nil {
+		body = s.source
+	} else {
+		src, err := s.searchSource.Source()
+
+		if err != nil {
+			return nil, err
+		}
+
+		// searchSource.Source() actually returns a map, yet its return type
+		// is specified as interface...
+		body = src.(map[string]interface{})
+	}
+
 	// Search
 	path += "/_search"
 
 	// Parameters
 	params := make(url.Values)
-	if !s.searchSource.hasSort() {
+	if _, ok := body["sort"]; !ok {
```

Comment:
It's not my approach, it's yours. You chose to make a check for something that relies on the DSL. The change I made above is the easiest way not to modify the behavior of ScanService when it is used without setting ScanService.Source()...

If I use Source I'm **not** on my own. There's no way to use Source on it's own because params depended on the DSL.

Agreed, remove the scan type.